vars/sectionDeployToEnvironment.groovy

```diff
@@ -33,29 +34,33 @@ def call(params) {
                 timeoutWithMsg(time: 120, unit: 'MINUTES', action: "buildinfra:${environment}") {
                   def additionalInfrastructureVariables = collectAdditionalInfrastructureVariablesFor(subscription, product, environment)
                   withEnv(additionalInfrastructureVariables) {
-                    tfOutput = spinInfra(product, component, environment, false, subscription)
+                    tfOutput = spinInfra(product, component, environment, tfPlanOnly, subscription)
                   }
                 }
               }
             }
 
-            registerDns(params)
+            if(!tfPlanOnly){
```

Comment:
This is a hack for time being, but this code definitely needs some cleanup at some point.